src/app/wallets/get-transactions-by-addresses.ts

```diff
@@ -0,0 +1,90 @@
+import { BTC_NETWORK, ONCHAIN_MIN_CONFIRMATIONS, SECS_PER_10_MINS } from "@config"
+
+import { getCurrentPrice } from "@app/prices"
+import { PartialResult } from "@app/partial-result"
+
+import { CacheKeys } from "@domain/cache"
+import { LedgerError } from "@domain/ledger"
+import { WalletTransactionHistory } from "@domain/wallets"
+import { OnChainError, TxDecoder, TxFilter } from "@domain/bitcoin/onchain"
+
+import { baseLogger } from "@services/logger"
+import { LedgerService } from "@services/ledger"
+import { RedisCacheService } from "@services/cache"
+import { OnChainService } from "@services/lnd/onchain-service"
+
+const redisCache = RedisCacheService()
+
+export const getTransactionsForWalletsByAddresses = async ({
+  wallets,
+  addresses,
+}: {
+  wallets: Wallet[]
+  addresses: OnChainAddress[]
```

Comment:
is there to check to verify `string[]` to `checkToOnChainAddress[]` is acceptable? ie: all onchain addresses should be valid addresses

Reply:
Technically we did it in the graphql layer as well inside the `OnChainAddress` scaler, but I think we can do it again inside the method for type consistency: https://github.com/GaloyMoney/galoy/pull/1511/commits/9bde789bef1186e64beab233afc72a90e585a1d3

Reply:
I actually think we don't need and could revert the last commit. I didn't know/remember we were already doing it on the graphql layer (I just look at the diff of this commit, didn't check the code outside of it). As we already do I think there is no need for double check imo.